src/hotspot/cpu/x86/x86.ad

```diff
@@ -9166,8 +9345,39 @@ instruct mask_all_evexL(kReg dst, rRegL src) %{
   format %{ "mask_all_evexL $dst, $src \t! mask all operation" %}
   ins_encode %{
     int vec_len = Matcher::vector_length(this);
-    __ kmovql($dst$$KRegister, $src$$Register);
-    __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
+    if (VM_Version::supports_avx512bw()) {
+      __ kmovql($dst$$KRegister, $src$$Register);
+      __ kshiftrql($dst$$KRegister, $dst$$KRegister, 64 - vec_len);
+    } else {
+      assert(vec_len <= 16, "");
+      __ kmovwl($dst$$KRegister, $src$$Register);
+      __ kshiftrwl($dst$$KRegister, $dst$$KRegister, 16 - vec_len);
+    }
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct mask_not_immLT8(kReg dst, kReg src, rRegI rtmp, kReg ktmp, immI_M1 cnt) %{
+  predicate(Matcher::vector_length(n) < 8 && VM_Version::supports_avx512dq());
+  match(Set dst (XorVMask src (MaskAll cnt)));
+  effect(TEMP_DEF dst, TEMP rtmp, TEMP ktmp);
+  format %{ "mask_not_LT8 $dst, $src, $cnt \t! mask not operation" %}
+  ins_encode %{
+    uint masklen = Matcher::vector_length(this);
+    __ knot(masklen, $dst$$KRegister, $src$$KRegister, $ktmp$$KRegister, $rtmp$$Register);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct mask_not_imm(kReg dst, kReg src, immI_M1 cnt) %{
+  predicate((Matcher::vector_length(n) == 8 && VM_Version::supports_avx512dq()) ||
+            (Matcher::vector_length(n) == 16) ||
+            (Matcher::vector_length(n) > 16 && VM_Version::supports_avx512bw()));
```

Comment:
What happens to not operation if vector_length == 8 and it is a !avx512dq platform?

Reply:
Same as above